llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -15710,48 +15710,51 @@ bool AArch64TargetLowering::shouldSinkOperands(
   return false;
 }
 
-static bool createTblShuffleForZExt(ZExtInst *ZExt, FixedVectorType *DstTy,
-                                    bool IsLittleEndian) {
-  Value *Op = ZExt->getOperand(0);
-  auto *SrcTy = cast<FixedVectorType>(Op->getType());
-  auto SrcWidth = cast<IntegerType>(SrcTy->getElementType())->getBitWidth();
-  auto DstWidth = cast<IntegerType>(DstTy->getElementType())->getBitWidth();
+static bool createTblShuffleMask(unsigned SrcWidth, unsigned DstWidth,
+                                 unsigned NumElts, bool IsLittleEndian,
+                                 SmallVectorImpl<int> &Mask) {
   if (DstWidth % 8 != 0 || DstWidth <= 16 || DstWidth >= 64)
     return false;
 
-  assert(DstWidth % SrcWidth == 0 &&
```

Comment:
Is the assert replaced by an `if` because it may trigger, possibly on an untested path?

Reply:
No, it just felt more like the conditions above rather than a compiler error / invariant violation that warrants an abort.

Reply:
Right, would be good to have a test case if this can happen

Reply:
Currently it can't happen, all the callers are with `i8` (i.e. `SrcWidth == 8`)

Reply:
Then keeping the code assert would seem appropriate, unless it is changes in the follow-up?